tests/unit/s2n_certificate_extensions_test.c

```diff
@@ -29,120 +29,53 @@
 #include "stuffer/s2n_stuffer.h"
 #include "utils/s2n_safety.h"
 
-#define SIZEOF_UINT24 3
+s2n_cert_public_key public_key;
+s2n_pkey_type actual_cert_pkey_type;
 
-static int s2n_certificate_extensions_send_empty(struct s2n_stuffer *stuffer)
-{
-    struct s2n_connection conn = { 0 };
-    return s2n_extension_list_send(S2N_EXTENSION_LIST_EMPTY, &conn, stuffer);
-}
+static const s2n_parsed_extension empty_extensions[S2N_PARSED_EXTENSIONS_COUNT] = { 0 };
 
-static int s2n_certificate_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *stuffer,
-        struct s2n_cert_chain_and_key *chain_and_key)
+#define EXPECT_EXTENSIONS_EMPTY(extensions) EXPECT_BYTEARRAY_EQUAL(extensions.parsed_extensions, empty_extensions, sizeof(empty_extensions))
+#define EXPECT_EXTENSIONS_NOT_EMPTY(extensions) EXPECT_BYTEARRAY_NOT_EQUAL(extensions.parsed_extensions, empty_extensions, sizeof(empty_extensions))
+
+static int s2n_skip_cert_chain_size(struct s2n_stuffer *stuffer)
 {
-    return s2n_extension_list_send(S2N_EXTENSION_LIST_CERTIFICATE, conn, stuffer);
+    uint32_t cert_chain_size;
+    GUARD(s2n_stuffer_read_uint24(stuffer, &cert_chain_size));
+    eq_check(cert_chain_size, s2n_stuffer_data_available(stuffer));
+    return S2N_SUCCESS;
 }
 
-static int s2n_certificate_extensions_parse(struct s2n_connection *conn, struct s2n_stuffer *stuffer)
+static int s2n_skip_cert(struct s2n_stuffer *stuffer)
 {
-    return s2n_extension_list_recv(S2N_EXTENSION_LIST_CERTIFICATE, conn, stuffer);
+    uint32_t cert_size;
+    GUARD(s2n_stuffer_read_uint24(stuffer, &cert_size));
+    GUARD(s2n_stuffer_skip_read(stuffer, cert_size));
+    return S2N_SUCCESS;
 }
 
-/* We can test total extension size based on the size of what we actually write */
-static int s2n_certificate_extensions_size(struct s2n_connection *conn, struct s2n_cert_chain_and_key *chain_and_key)
+static int s2n_x509_validator_validate_cert_chain_test(struct s2n_connection *conn, struct s2n_stuffer *stuffer)
 {
-    DEFER_CLEANUP(struct s2n_stuffer stuffer, s2n_stuffer_free);
-    GUARD(s2n_stuffer_growable_alloc(&stuffer, 0));
+    GUARD(s2n_skip_cert_chain_size(stuffer));
+    uint32_t cert_chain_size = s2n_stuffer_data_available(stuffer);
+
+    uint8_t *cert_chain_data;
+    notnull_check(cert_chain_data = s2n_stuffer_raw_read(stuffer, cert_chain_size));
 
-    struct s2n_connection conn_copy = *conn;
-    GUARD(s2n_certificate_extensions_send(&conn_copy, &stuffer, NULL));
-    int size = s2n_stuffer_data_available(&stuffer);
+    GUARD(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn,
```

Comment:
Since this function returns `s2n_cert_validation_code`, we could explicit check for the return value instead of a `GUARD` here.

Reply:
GUARD should cover it. The `s2n_cert_validation_code` is just "ok, or specific error", and we don't care about the specific error. It must succeed for the test to pass.